apps/web/src/studio/components/workflows/step-editor/WorkflowsStepEditorPage.tsx

```diff
@@ -29,11 +33,27 @@ export const WorkflowsStepEditorPage = () => {
   });
   const step = workflow?.steps.find((item) => item.stepId === stepId);
   const title = step?.stepId;
+  const stepType: ChannelTypeEnum | undefined = step?.type;
 
-  const navigate = useNavigate();
-  const handleTestClick = () => {
-    // TODO: this is just a temporary step for connecting the prototype
-    navigate(ROUTES.STUDIO_FLOWS_TEST_STEP);
+  const { mutateAsync: testSendEmailEvent, isLoading: isTestingEmail } = useMutation(testSendEmailMessage);
+
+  const handleTestClick = async () => {
+    try {
+      await testSendEmailEvent({
+        stepId,
+        workflowId: workflow.workflowid,
```

Comment:
Why do we have this discrepancy with casing on the `id`? We should eliminate it.

```suggestion
        workflowId: workflow.workflowId,
```